packages/core/src/browser/keybinding.ts

```diff
@@ -416,14 +416,16 @@ export class KeybindingRegistry {
      * @param binding to execute
      * @param event keyboard event.
      */
-    protected executeKeyBinding(binding: Keybinding, event: KeyboardEvent): void {
+    protected executeKeyBinding(binding: common.Keybinding, event: KeyboardEvent): void {
         if (this.isPseudoCommand(binding.command)) {
             /* Don't do anything, let the event propagate.  */
         } else {
             const command = this.commandRegistry.getCommand(binding.command);
             if (command) {
-                this.commandRegistry.executeCommand(binding.command, binding.args)
-                    .catch(e => console.error('Failed to execute command:', e));
+                if (this.commandRegistry.isEnabled(binding.command, binding.args)) {
```

Comment:
I only wonder whether it is a proper place. We have a method `isEnabled` in this class, which actually used to match keybindings. I am not sure whether it should be moved there? i.e. if keybinding is matching, but there is no active handler, should we skip keybinding and try next? With current proposal we pick keybinding and do nothing. There is also `isActive` method which is not used anymore. Maybe during refactoring we lost it and the actual fix is to call `isActive` from `isEnabled` if keybinding is matching.

Reply:
Yes, this is fair reasoning.

> With current proposal we pick keybinding and do nothing

Well, with the `master` state, we pick the keybinding, and we equally do nothing as an error is thrown.

With the proposed change:
 - we do not fire the `fireWillExecuteCommand` if there are no active handlers. (same on `master`)
 - we `preventDefault` and `stopPropagation` no matter there were any active handlers or not. (same as on `master`)

The only behavioral change is that we do not throw an error.

> Maybe during refactoring we lost it

OK, I will look into the commit history.

Reply:
ok, let's try